docs/site/Booting-an-Application.md

```diff
@@ -259,12 +259,118 @@ The options for this are passed in a `services` object on `BootOptions`.
 
-| Options      | Type                 | Default           | Description                                                                                                  |
-| ------------ | -------------------- | ----------------- | ------------------------------------------------------------------------------------------------------------ |
-| `dirs`       | `string \| string[]` | `['services']`    | Paths relative to projectRoot to look in for Service artifacts                                               |
-| `extensions` | `string \| string[]` | `['.service.js']` | File extensions to match for Service artifacts                                                               |
-| `nested`     | `boolean`            | `true`            | Look in nested directories in `dirs` for Service artifacts                                                   |
-| `glob`       | `string`             |                   | A `glob` pattern string. This takes precedence over above 3 options (which are used to make a glob pattern). |
+| Options      | Type                 | Default           | Description                                                                                                           |
+| ------------ | -------------------- | ----------------- | --------------------------------------------------------------------------------------------------------------------- |
+| `dirs`       | `string \| string[]` | `['services']`    | Paths relative to projectRoot to look in for Service artifacts                                                        |
+| `extensions` | `string \| string[]` | `['.service.js']` | File extensions to match for Service artifacts                                                                        |
+| `nested`     | `boolean`            | `true`            | Look in nested directories in `dirs` for Service artifacts                                                            |
+| `globs`      | `string \| string[]` |                   | A `glob` pattern string or array. This takes precedence over above 3 options (which are used to make a glob pattern). |
+
+### Configuration Booter
+
+#### Description
+
+Discovers configuration files in `configs` and `dist/configs` folders and binds
+them to the application context using `app.configure()`.
+
+The booter also serves as an extension point to plug in configuration loaders
+that handle different forms of files. By default, a configuration loader is
+added to recognize `.config.js`, `.config.json`, `.config.yaml`, and
+`.config.yml` files. Files with the same base name are loaded in the precedence
+of js, json, yaml, and yml extensions. For example, `x.config.js` shadows
+`x.config.json`.
+
+The js/json/yaml/yml files export an object containing binding key/configuration
+value pairs.
+
+The `.ts` file will be transpiled to `.js`.
+
+```ts
+export = {
+  'loggers.Log1': {
+    level: 'info',
+    __source: 'js',
```

Comment:
What is the purpose of `__source` property and why do users have to specify it explicitly? I would expect the framework to take care of that. What happens when the user makes a mistake and either forgets to add `__source` property or provides a wrong value - e.g. `ts` instead of `js`, because that feels more natural?

Reply:
> The `.ts` file

Which `.ts` file?

Agreed, `__source ` does not feel intuitive.

Reply:
Agreed, `__source ` does not feel intuitive. Also, it is a single underscore on other files.

Reply:
The real purpose of `_source` is for testing as we can assert the value is loaded from the expected source.